pds_doi_core/actions/check.py

```diff
@@ -0,0 +1,409 @@
+                    msg["Subject"]  = subject_field
+                    msg["To"]       = final_receivers
+
+                    msg.set_content(email_entire_message)
+
+                    (attachment_filename,attachment_part) = self._prepare_attachment(o_dicts_per_submitter)
+                    msg.add_attachment(attachment_part)  # The attachment is now 'attached' in the msg object.
+
+                    # Send the email with attachment file.
+                    self._emailer.send_message(msg)
+
+                    # Delete the temporary attached file.
+                    if os.path.isfile(attachment_filename):
+                        os.remove(attachment_filename) # Remove temporary file.
+        return 1
+
+    def run(self,
+            output_format='JSON', query_criterias=[], to_send_mail_flag=True, to_send_attachment_flag=True):
+        """
+        Function query the local database for latest records for pending state and check OSTI server all the records with criteria specified in query_criterias return the object either in JSON or XML.
+        Once the query is returned every record will be checked for initial status and status returned from OSTI.
+        If the status has changed from initial status, write a new record to the database.
+        All parameters are optional and may be useful for tests.
+        :param output_format:
+        :param query_criterias:
+        :param to_send_mail_flag:
+        :param to_send_attachment_flag:
+        :return: o_check_result:
+        """
+
+        logger.debug(f"to_send_mail_flag,output_format {to_send_mail_flag,output_format}")
+        o_check_result = []
+
+        # Get the list of latest rows in database with status = 'Pending'.
+        if len(query_criterias) > 0: 
+            # Use query_criterias if provided from user.
+            o_doi_list = self._list_obj.run(output_format=output_format,query_criterias=query_criterias)
+        else:
+            # Get the list of latest rows in database with status = 'Pending' using the private self._query_criterias.
+            o_doi_list = self._list_obj.run(output_format=output_format,query_criterias=self._query_criterias)
+
+        # Convert from JSON into a list.
+        pending_state_list = []
+        if len(o_doi_list) > 0:
+            pending_state_list = json.loads(o_doi_list)
+
+        # Variable pending_state_list is now list of records with 'Pending status.
+
+        logger.debug(f"pending_state_list {pending_state_list,len(pending_state_list)}")
+
+        # For every doi_value in doi_list, make a query to the server for the status and update the state in the database.
+
+        i_url = self._config.get('OSTI', 'url')
+
+        for pending_record in pending_state_list:
+            logger.debug(f"pending_record {pending_record}")
+            doi_value = pending_record['doi']
+            query_dict = {'doi':doi_value}
+            if 'start_update' in self._query_criterias:
```

Comment:
Because we have removed the optional argument for 'check' action, there is no longer any need to add 'start_update' and 'end_update' to the query criteria.

Reply:
ok good.